base/reflection.jl

```diff
@@ -233,28 +233,51 @@ function methods(f::ANY)
     end
 end
 
+function visit(f, mt::MethodTable)
```

Comment:
I don't see why this is an improvement over being able to iterate over a list of methods. Also `methods` still returns an array sometimes, so now its result is sometimes iterable and sometimes not.